.github/workflows/pytest.yml

```diff
@@ -65,17 +67,45 @@ jobs:
           python-version: ${{ matrix.python-version }}
           cache: pip
           cache-dependency-path: 'requirements/*.txt'
+      - name: Cache python packages
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ hashFiles('requirements/base.txt') }}-${{ hashFiles('requirements.txt/test.txt') }}
       - name: Install Python dependencies
         run: make install-python-test
       - name: Install Node
         uses: actions/setup-node@v3
         with:
           node-version: latest
           cache: npm
+      - name: Cache node modules
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ matrix.os }}-${{ matrix.python-version }}-node_modules-${{ hashFiles('package-lock.json') }}
```

Comment:
Why is the Python version a factor here? For parallelized access to cache?

Reply:
The 2 tests are running in parallel and their caches should be maintained separately, else the cache for one would be overwritten by another. Even if the content of the caches generated by both are the same, this is an area of uncertainty IMO, depending on how caches are being maintained behind the scenes.

Reply:
Will it work to use just `{{ matrix }}`? Let all the sub-vars be used then.

Reply:
I am not sure if `{{ matrix }}` is accepted by GHA? Should I test it out?

Reply:
Yes. There may be a way to get a flattened representation too.

Reply:
For flattened representation, use `${{ toJSON(matrix) }}` https://docs.github.com/en/actions/learn-github-actions/expressions#tojson